lib/CMakeLists.txt

```diff
@@ -78,32 +75,23 @@ if (SWIFT_SWIFT_PARSER)
   # toolchain.
   list(TRANSFORM SWIFT_SYNTAX_MODULES APPEND ".swiftmodule"
        OUTPUT_VARIABLE SWIFT_SYNTAX_MODULE_DIRS)
+
+  # Interface library to collect swiftinterfaces and swiftmodules from
+  # SwiftSyntax
+  add_library(swiftSyntaxLibraries INTERFACE)
   foreach(module_dir ${SWIFT_SYNTAX_MODULE_DIRS})
     # Find all of the source module files.
     file(GLOB module_files
          "${SWIFT_SYNTAX_LIBRARIES_SOURCE_DIR}/${module_dir}/*.swiftinterface")
 
-    # Determine the destination module files.
-    set(dest_module_files)
-    foreach(full_module_file ${module_files})
-        get_filename_component(module_file ${full_module_file} NAME)
-        list(APPEND dest_module_files
-            "${SWIFT_HOST_LIBRARIES_DEST_DIR}/${module_dir}/${module_file}")
-    endforeach()
```

Comment:
I'm pretty sure we still want this + add `OUTPUT ${dest_module_files}` to the new custom target. We also need to do this for the dylib.